distributed/client.py

```diff
@@ -149,6 +150,14 @@ def _del_global_client(c: Client) -> None:
             pass
 
 
+class WorkerWaitMode(Enum):
```

Comment:
I'm anti-enums for a user-facing API. If you feel strongly about it, go head, but it's always annoying as a user to have to figure out where to import some enum from in a package in order to do a very simple thing like this. `typing.Literal` is a great way to document and statically enforce the valid values without requiring users to go through the inconvenience of another import.

EDIT: I see you have
```
        if isinstance(mode, str):
            mode = WorkerWaitMode(mode) 
```
for convenience. IMO, an enum is still overkill though.

Reply:
I agree with the anti-enum stance, especially now that `Literal`s are widely supported. `Literal`s even have some additional features that enums can't match like unions of different subtypes.

Reply:
Neat! I haven't used `typing.Literal` yet but it definitely sounds better, it requires Python 3.8 but I think we should be good here with that. I'll refactor and drop the `enum`.